floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -1,14 +1,46 @@
+import copy
 import csv
 
 from floodmodeller_api import DAT
 
 
+def extract_attrs(object_input, keys):
```

Comment:
I like that you extracted this as a function - I think it helps with readability